test/cmake/add_subdirectories.cmake

```diff
@@ -2,8 +2,6 @@
 # SPDX-FileCopyrightText: 2016-2024 Knut Reinert & MPI für molekulare Genetik
 # SPDX-License-Identifier: BSD-3-Clause
 
-cmake_minimum_required (VERSION 3.10...3.30)
```

Comment:
What is the reasoning to sometimes adapt the versions and sometimes to just remove them?

Reply:
It's required for CMakeLists.txt.
If it's a script file like here, you only need it if you use that script file in script mode (`cmake - P`).
If it's included from a CMakeLists.txt, the required minimum version is already established.

Having said that, we can also keep it everywhere. 
The only drawback is that we need to touch more files to bump the version
The advantage would be that we don't need to care about how a script file is used. 

What do you prefer?

Reply:
Yes, I was just referring to the fact that further down there seems to be a bunch of script files where it could have been removed as well? So I was curious if there is something more to it.
I'd prefer removing it everywhere where it is not necessary, simply to avoid having many places where there could be inconsistencies in the future.

Reply:
I removed it everywhere possible.
It's required for all the `CMakeLists.txt`.
It's removed from all `*.cmake` files, except `generate_include_dependencies.cmake` because this one is used in script mode `cmake -P` where it then basically acts as an `CMakeLists.txt`.